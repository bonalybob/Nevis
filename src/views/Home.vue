<template>
  <div>
    <div class="bg-gradient text-center w-full shadow-md mb-3">
      <img src="@/assets/images/Nevis Logo.png" alt="Nevis Logo" class="h-40 py-5" />
      <h1 class="pb-4 text-3xl text-white">Welcome to Nevis</h1>
    </div>
    <div class="mx-10 mb-3">
      <button class="button" @click="connect">Connect</button>
      <router-link tag="button" class="button" to="/about">About</router-link>
    </div>
    <div class="shadow mx-10">
      <text-input v-model.trim="server" label="Server:" />
      <text-input v-model.trim="port" label="Port:" />
      <text-input v-model.trim="username" label="Username:" />
      <text-input v-model.trim="password" label="Password:" type="password" />
      <text-input v-model.trim="database" label="Database:" />
    </div>
  </div>
</template>

<script>
import TextInput from '@/components/TextInput'

export default {
  components: {
    'text-input': TextInput,
  },

  data: function () {
    return {
      server: 'localhost',
      port: '3306',
      username: 'sa',
      password: 'orienteer',
      database: 'nevis',
    }
  },

  methods: {
    connect: function () {
      this.$database.connection = this.$mysql.createConnection({
        connectionLimit: 10,
        host: this.server,
        port: this.port,
        user: this.username,
        password: this.password,
        database: this.database,
      })
      return this.$database.connect()
        .then(() => {
          this.$database.connected = true
          this.$router.push('/events')
        })
        .catch(error => this.$messages.addMessage(error, 'error'))
    },
  },
}
</script>
