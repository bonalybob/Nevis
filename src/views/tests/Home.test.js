import { mount, shallowMount } from '@vue/test-utils'
import Home from '@/views/Home'

test('Is a Vue Instance', () => {
  const wrapper = shallowMount(Home, {
    stubs: ['router-link'],
  })
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = shallowMount(Home, {
    stubs: ['router-link'],
  })
  expect(wrapper.element).toMatchSnapshot()
})

test('Connect to Database - Success', async () => {
  const mockCreateConnection = jest.fn()
  const mockConnect = jest.fn()
  const mockAddMessage = jest.fn()
  const mockRouterPush = jest.fn()
  mockConnect.mockResolvedValue()

  const wrapper = mount(Home, {
    stubs: ['router-link'],
    mocks: {
      $database: { connection: {}, connect: mockConnect, connected: false },
      $mysql: { createConnection: mockCreateConnection },
      $messages: { addMessage: mockAddMessage },
      $router: { push: mockRouterPush },
    },
  })
  await wrapper.vm.connect()
  expect(mockCreateConnection).toHaveBeenCalledTimes(1)
  expect(mockConnect).toHaveBeenCalledTimes(1)
  expect(mockAddMessage).toHaveBeenCalledTimes(0)
  expect(wrapper.vm.$database.connected).toBeTruthy()
  expect(mockRouterPush).toHaveBeenCalledTimes(1)
  expect(mockRouterPush).toHaveBeenLastCalledWith('/events')
})

test('Connect to Database - Error', async () => {
  const mockCreateConnection = jest.fn()
  const mockConnect = jest.fn()
  const mockAddMessage = jest.fn()
  mockConnect.mockRejectedValue('Error')

  const wrapper = mount(Home, {
    stubs: ['router-link'],
    mocks: {
      $database: { connection: {}, connect: mockConnect, connected: false },
      $mysql: { createConnection: mockCreateConnection },
      $messages: { addMessage: mockAddMessage },
    },
  })
  await wrapper.vm.connect()
  expect(mockCreateConnection).toHaveBeenCalledTimes(1)
  expect(mockConnect).toHaveBeenCalledTimes(1)
  expect(mockAddMessage).toHaveBeenCalledTimes(1)
  expect(mockAddMessage).toHaveBeenLastCalledWith('Error', 'error')
  expect(wrapper.vm.$database.connected).toBeFalsy()
})
