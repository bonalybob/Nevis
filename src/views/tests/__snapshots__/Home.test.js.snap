// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div>
  <div
    class="bg-gradient text-center w-full my-shadow-lg mb-5"
  >
    <img
      alt="Nevis Logo"
      class="h-40 py-5"
      src="@/assets/images/Nevis Logo.png"
    />
     
    <h1
      class="pb-4 text-3xl text-white"
    >
      Welcome to Nevis
    </h1>
  </div>
   
  <div
    class="mx-10 mb-3"
  >
    <button
      class="button"
    >
      Connect
    </button>
     
    <router-link-stub
      class="button"
      tag="button"
      to="/about"
    >
      About
    </router-link-stub>
     
    <router-link-stub
      class="button"
      tag="button"
      to="/help"
    >
      Help
    </router-link-stub>
  </div>
   
  <div
    class="my-shadow mx-10"
  >
    <text-input-stub
      label="Server:"
      type=""
      value="localhost"
    />
     
    <text-input-stub
      label="Port:"
      type=""
      value="3306"
    />
     
    <text-input-stub
      label="Username:"
      type=""
      value="sa"
    />
     
    <text-input-stub
      label="Password:"
      type="password"
      value="orienteer"
    />
     
    <text-input-stub
      label="Database:"
      type=""
      value="nevis"
    />
  </div>
</div>
`;
